dashboard_api/db/static/sites/__init__.py

```diff
@@ -2,53 +2,73 @@
 import os
 from enum import Enum
 from typing import List
+import json
 
+import botocore
+
+from dashboard_api.db.utils import s3_get
+from dashboard_api.models.static import Sites, Link
+from dashboard_api.core.config import (SITE_METADATA_FILENAME, BUCKET)
 from dashboard_api.db.static.errors import InvalidIdentifier
 from dashboard_api.db.utils import get_indicators, indicator_exists, indicator_folders
 from dashboard_api.models.static import Site, Sites
 
-data_dir = os.path.join(os.path.dirname(__file__))
-
-
 class SiteManager(object):
     """Default Site holder."""
 
     def __init__(self):
-        """Load all datasets in a dict."""
-        site_ids = [
-            os.path.splitext(f)[0] for f in os.listdir(data_dir) if f.endswith(".json")
-        ]
+        """Load all sites."""
+        pass
 
-        self._data = {
-            site: Site.parse_file(os.path.join(data_dir, f"{site}.json"))
-            for site in site_ids
-        }
-
-    def get(self, identifier: str) -> Site:
+    # todo: needs caching
+    def get(self, identifier: str, api_url: str) -> Site:
         """Fetch a Site."""
         try:
-            site = self._data[identifier]
+            sites = self.get_all(api_url)
+            site = next(filter(lambda x: x.id == identifier, sites.sites), None)
             site.indicators = get_indicators(identifier)
             return site
         except KeyError:
             raise InvalidIdentifier(f"Invalid identifier: {identifier}")
 
-    def get_all(self) -> Sites:
+    def get_all(self, api_url: str) -> Sites:
         """Fetch all Sites."""
-        all_sites = [site.dict() for site in self._data.values()]
-        indicators = indicator_folders()
-        # add indicator ids
-        for site in all_sites:
-            site["indicators"] = [
-                ind for ind in indicators if indicator_exists(site["id"], ind)
-            ]
-        return Sites(sites=all_sites)
+        if os.environ.get('ENV') == 'local':
+            # Useful for local testing
+            example_sites = "example-site-metadata.json"
+            print(f'Loading {example_sites}')
+            s3_datasets = json.loads(open(example_sites).read())
+        try:
+            print(f"{BUCKET}/{SITE_METADATA_FILENAME}")
+            s3_datasets = json.loads(
+                s3_get(bucket=BUCKET, key=SITE_METADATA_FILENAME)
+            )
+            print("sites json successfully loaded from S3")
 
-    def list(self) -> List[str]:
-        """List all sites"""
-        return list(self._data.keys())
+            # indicators = indicator_folders()
```

Comment:
I think we can remove this code

Reply:
I was going to ask about this -- I wasn't sure what the indicators were, and they're empty in all the examples I saw. I thought it might be something specific to the covid dashboard. Should I remove all the indicators code (instead of just this one place)?

Reply:
Actually, I just fixed the indicators code to work, though I don't have any examples to test with.